src/daycare.c

```diff
@@ -174,26 +193,42 @@ static void TransferEggMoves(void)
 {
     u32 i, j, k, l;
     u16 numEggMoves;
-    struct Pokemon mon;
 
     for (i = 0; i < DAYCARE_MON_COUNT; i++)
     {
+        u16 moveLearnerSpecies = GetBoxMonData(&gSaveBlock1Ptr->daycare.mons[i].mon, MON_DATA_SPECIES);
+        u16 eggSpecies = GetEggSpecies(moveLearnerSpecies);
+
         if (!GetBoxMonData(&gSaveBlock1Ptr->daycare.mons[i].mon, MON_DATA_SANITY_HAS_SPECIES))
             continue;
 
-        BoxMonToMon(&gSaveBlock1Ptr->daycare.mons[i].mon, &mon);
+        // Make sure Snorlax gets Snorlax's egg moves instead of Munchlax's
```

Comment:
I would generalise this comment to refer to Incense babies or something similar rather than specifically mentioning a case

Reply:
Shouldn't this also specifically rely on the incense config?

Reply:
Nope, they have seperate egg moves even in gen 9.

Reply:
Actually I might be wrong here

Reply:
I don't know how to handle this, mirror herb and incense breeding don't exist in the same game.
The expansion also has egg moves based on the egg move lists of every generation combined, while gen 9 games have the same ones for the entire evolution line.
Perhaps it would make sense to directly change their egg moves based on the config's setting.